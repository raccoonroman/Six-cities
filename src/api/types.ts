import { OfferCategory, Location } from '@/types';

export interface AuthInfoRaw {
  avatar_url: string;
  email: string;
  id: number;
  is_pro: boolean;
  name: string;
}

export interface User {
  email: string;
  password: string;
}

export interface OfferRaw {
  id: number;
  title: string;
  preview_image: string;
  price: number;
  rating: number;
  type: OfferCategory;
  bedrooms: number;
  max_adults: number;
  is_favorite: boolean;
  is_premium: boolean;
  location: Location;
  city: {
    name: string;
    location: Location;
  };
  description: string;
  goods: string[];
  host: {
    avatar_url: string;
    id: number;
    is_pro: boolean;
    name: string;
  };
  images: string[];
}

export interface CommentRaw {
  id: number;
  comment: string;
  date: string;
  rating: number;
  user: {
    avatar_url: string;
    id: number;
    is_pro: boolean;
    name: string;
  };
}

export interface CommentPost {
  comment: string;
  rating: number;
}
