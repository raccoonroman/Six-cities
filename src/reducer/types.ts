import { OfferRaw } from '@/api/types';
import { ActionType } from '@/const';

interface LoadOffersAction {
  type: typeof ActionType.LOAD_OFFERS
  payload: OfferRaw[]
}

interface UpdateOfferAction {
  type: typeof ActionType.UPDATE_OFFER
  payload: OfferRaw
}

interface SetEmailAction {
  type: typeof ActionType.SET_EMAIL
  payload: string
}

export type offersAction = LoadOffersAction | UpdateOfferAction;
export type userDataAction = SetEmailAction;
