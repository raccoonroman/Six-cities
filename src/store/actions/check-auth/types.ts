enum CheckAuth {
  PENDING = 'CHECK_AUTH/PENDING',
  RESOLVE = 'CHECK_AUTH/RESOLVE',
  REJECT = 'CHECK_AUTH/REJECT',
}

export default CheckAuth;
