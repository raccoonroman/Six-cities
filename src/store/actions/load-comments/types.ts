enum LoadComments {
  PENDING = 'LOAD_COMMENTS/PENDING',
  RESOLVE = 'LOAD_COMMENTS/RESOLVE',
  REJECT = 'LOAD_COMMENTS/REJECT',
}

export default LoadComments;
