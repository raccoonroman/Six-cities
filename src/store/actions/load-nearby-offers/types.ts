enum LoadNearbyOffers {
  PENDING = 'LOAD_NEARBY_OFFERS/PENDING',
  RESOLVE = 'LOAD_NEARBY_OFFERS/RESOLVE',
  REJECT = 'LOAD_NEARBY_OFFERS/REJECT',
}

export default LoadNearbyOffers;
