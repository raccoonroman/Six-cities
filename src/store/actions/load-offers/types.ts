enum LoadOffers {
  PENDING = 'LOAD_OFFERS/PENDING',
  RESOLVE = 'LOAD_OFFERS/RESOLVE',
  REJECT = 'LOAD_OFFERS/REJECT',
}

export default LoadOffers;
