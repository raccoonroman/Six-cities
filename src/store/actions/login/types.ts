enum Login {
  PENDING = 'LOGIN/PENDING',
  RESOLVE = 'LOGIN/RESOLVE',
  REJECT = 'LOGIN/REJECT',
}

export default Login;
