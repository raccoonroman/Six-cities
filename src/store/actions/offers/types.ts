export enum LoadOffers {
  PENDING = 'LOAD_OFFERS/PENDING',
  RESOLVE = 'LOAD_OFFERS/RESOLVE',
  REJECT = 'LOAD_OFFERS/REJECT',
}

export enum UpdateOffer {
  PENDING = 'UPDATE_OFFER/PENDING',
  RESOLVE = 'UPDATE_OFFER/RESOLVE',
  REJECT = 'UPDATE_OFFER/REJECT',
}
