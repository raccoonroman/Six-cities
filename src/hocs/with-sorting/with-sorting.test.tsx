import React from 'react';
import renderer from 'react-test-renderer';
import { Offer } from '@/types';
import withSorting from '@/hocs/with-sorting';

const mockOffers: Offer[] = [
  {
    id: 336,
    title: 'Title 1',
    previewImage: 'https://i.picsum.photos/id/336/260/200.jpg',
    price: 13,
    rating: 4.9,
    type: 'house',
    bedrooms: 3,
    maxAdults: 3,
    isFavorite: true,
    isPremium: false,
    location: {
      latitude: 52.359160,
      longitude: 4.849366,
      zoom: 12,
    },
    city: {
      name: 'Vinnytsia',
      location: {
        latitude: 52.370216,
        longitude: 4.895168,
        zoom: 13,
      },
    },
  },
  {
    id: 337,
    title: 'Title 2',
    previewImage: 'https://i.picsum.photos/id/337/260/200.jpg',
    price: 400,
    rating: 4.2,
    type: 'room',
    bedrooms: 1,
    maxAdults: 2,
    isFavorite: false,
    isPremium: true,
    location: {
      latitude: 52.367503,
      longitude: 4.752484,
      zoom: 12,
    },
    city: {
      name: 'Vinnytsia',
      location: {
        latitude: 52.370216,
        longitude: 4.895168,
        zoom: 13,
      },
    },
  },
  {
    id: 338,
    title: 'Title 3',
    previewImage: 'https://i.picsum.photos/id/338/260/200.jpg',
    price: 456,
    rating: 3.8,
    type: 'apartment',
    bedrooms: 4,
    maxAdults: 6,
    isFavorite: false,
    isPremium: false,
    location: {
      latitude: 52.501544,
      longitude: 4.058915,
      zoom: 12,
    },
    city: {
      name: 'Vinnytsia',
      location: {
        latitude: 52.370216,
        longitude: 4.895168,
        zoom: 13,
      },
    },
  },
];

const currentCity = 'Vinnytsia';

interface Props {
  className: string;
}

const MockComponent: React.FC<Props> = ({ className }) => (
  <div className={className} />
);

const MockComponentWrapped = withSorting(MockComponent);

it('Render <withSorting />', () => {
  const tree = renderer.create(
    <MockComponentWrapped
      offers={mockOffers}
      currentCity={currentCity}
    />,
  ).toJSON();

  expect(tree).toMatchSnapshot();
});
