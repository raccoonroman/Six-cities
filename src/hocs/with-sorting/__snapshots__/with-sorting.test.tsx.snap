// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render <withSorting /> 1`] = `
<section
  className="cities__places places"
>
  <h2
    className="visually-hidden"
  >
    Places
  </h2>
  <b
    className="places__found"
  >
    3
     
    places to stay in
     
    Vinnytsia
  </b>
  <form
    action="#"
    className="places__sorting"
    method="get"
  >
    <span
      className="places__sorting-caption"
    >
      Sort by 
    </span>
    <span
      className="places__sorting-type"
      onClick={[Function]}
      onKeyDown={[Function]}
      role="button"
      tabIndex={0}
    >
      Popular
      <svg
        className="places__sorting-arrow"
        height="4"
        width="7"
      >
        <use
          xlinkHref="#icon-arrow-select"
        />
      </svg>
    </span>
    <ul
      className="places__options places__options--custom"
    >
      <li
        className="places__option places__option--active"
        onClick={[Function]}
        onKeyDown={[Function]}
      >
        Popular
      </li>
      <li
        className="places__option"
        onClick={[Function]}
        onKeyDown={[Function]}
      >
        Price: low to high
      </li>
      <li
        className="places__option"
        onClick={[Function]}
        onKeyDown={[Function]}
      >
        Price: high to low
      </li>
      <li
        className="places__option"
        onClick={[Function]}
        onKeyDown={[Function]}
      >
        Top rated first
      </li>
    </ul>
  </form>
  <div
    className="cities__places-list places__list"
  />
</section>
`;
